cpp/modmesh/io/gmsh.hpp

```diff
@@ -0,0 +1,397 @@
+#pragma once
+#include <string>
+#include <fstream>
+#include <vector>
+#include <unordered_map>
+#include <functional>
+
+#include <modmesh/base.hpp>
+#include <modmesh/mesh/mesh.hpp>
+#include <modmesh/buffer/buffer.hpp>
+
+namespace modmesh
+{
+namespace IO
```

Comment:
namespace should not use uppercase, and it should not have the unnecessary ambiguity to the Python module named `io`. 

In addition, `IO` is a pretty short name and may conflict with symbols in foreign code base.

Please rename it to `inout`.

Reply:
I renamed the module name to `inout` in https://github.com/solvcon/modmesh/pull/233/commits/7432534167ee1da998cfb1a7024ed5e8f52f869f too.